{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Canvas Creation from Enrichr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import json\n",
    "import requests\n",
    "import math\n",
    "import matplotlib\n",
    "import uuid\n",
    "from textwrap import dedent\n",
    "from IPython.core.display import display, HTML\n",
    "from string import Template\n",
    "from random import seed, randint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below this is arbitrary definitions for these variables, their actual values will be selected in the Appyter (so DON'T ADD BELOW CELL TO APPYTER)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "bar_color = 'mediumforestgreen'\n",
    "bar_color_not_sig = 'lightgrey'\n",
    "edgecolor = None\n",
    "chart_type = \"Canvas\"\n",
    "all_libraries = ['WikiPathways_2019_Human']\n",
    "color = 'Pink'\n",
    "scaling_factor = ''\n",
    "\n",
    "# list of genes that was input\n",
    "\n",
    "open_gene_list_file = open('geneList.txt','r')\n",
    "lines = open_gene_list_file.readlines()\n",
    "genes = [x.strip().upper() for x in lines]\n",
    "open_gene_list_file.close()\n",
    "\n",
    "#genes = ['TP53', 'TNF', 'EGFR', 'GKN1', 'HADHA', 'APOE', 'ESR1', 'VEGFA', 'TGFB1', 'PREPL', 'TIA1', 'TPO', 'TTN', 'SATB2', 'CHPF', 'MALL', 'MIPIP', 'NUPL1', 'IL6', 'PDIA3', 'CTNNB1', 'SLC39A1', 'DTNA','SLC1A1', 'GALNT2', 'HIST2H2AC', 'CD63']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "tags": [
     "outputPrepend"
    ]
   },
   "outputs": [],
   "source": [
    "def library_processing(index):\n",
    "    '''processes library data. returns anneal_list, x_dimension, y_dimension, num_hex'''\n",
    "    raw_library_data = []\n",
    "    library_data = []\n",
    "\n",
    "    library_name = all_libraries[index]\n",
    "    with open('Enrichr-Libraries/'+library_name, 'r') as f:\n",
    "        for line in f.readlines():\n",
    "            raw_library_data.append(line.split(\"\\t\\t\"))\n",
    "    name = []\n",
    "    gene_list = []\n",
    "\n",
    "    for i in range(len(raw_library_data)):\n",
    "        name += [raw_library_data[i][0]]\n",
    "        raw_genes = raw_library_data[i][1].split(\"\\t\")\n",
    "        gene_list += [raw_genes[:-1]]\n",
    "\n",
    "    library_data = [list(a) for a in zip(name, gene_list)]\n",
    "\n",
    "    # library_data: a 2D list where the first element is the name and the second element is a list      of genes associated with that name\n",
    "\n",
    "    jaccard_indices = []\n",
    "    indices = []\n",
    "\n",
    "    for gene_set in library_data:\n",
    "        intersection = [value for value in gene_set[1] if value in genes]\n",
    "        index = len(intersection)/(len(gene_set[1]) + len(genes))\n",
    "        jaccard_indices += [[gene_set[0], index]]\n",
    "        indices += [round(index, 5)]\n",
    "\n",
    "    # determine the dimensions of the canvas\n",
    "    x_dimension = math.ceil(math.sqrt(len(indices)))\n",
    "    y_dimension = math.ceil(math.sqrt(len(indices)))\n",
    "\n",
    "    # Zip name, gene_list, indices, and blank list for neighbor score then add dummy entries to the      zipped list\n",
    "    num_hex = x_dimension*y_dimension\n",
    "    neighbor_score = [0.0] * len(name)\n",
    "    anneal_list = list(zip(name, gene_list, indices, neighbor_score))\n",
    "\n",
    "    unzipped_anneal_list = [name, gene_list, indices, neighbor_score]\n",
    "\n",
    "    # Add \"dummy\" hexagons\n",
    "    for i in range(len(name), num_hex):\n",
    "        anneal_list += [('', [], 0.0, 0.0)]\n",
    "\n",
    "    return anneal_list, x_dimension, y_dimension, num_hex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_color(anneal_list):\n",
    "    r_value = 0\n",
    "    g_value = 0\n",
    "    b_value = 0\n",
    "\n",
    "    if color == 'Red':\n",
    "        r_value = 1.0\n",
    "        g_value = 0.3\n",
    "        b_value = 0.3\n",
    "    if color == 'Blue':\n",
    "        r_value = 0.0\n",
    "        g_value = 0.3\n",
    "        b_value = 1.0\n",
    "    if color == 'Purple':\n",
    "        r_value = 0.5\n",
    "        g_value = 0.0\n",
    "        b_value = 1.0\n",
    "    if color == 'Green':\n",
    "        r_value = 0.0\n",
    "        g_value = 1.0\n",
    "        b_value = 0.2\n",
    "    if color == 'Grey':\n",
    "        r_value = 1.0\n",
    "        g_value = 1.0\n",
    "        b_value = 1.0\n",
    "    if color == 'Orange':\n",
    "        r_value = 1.0\n",
    "        g_value = 0.55\n",
    "        b_value = 0.0\n",
    "    if color == 'Pink':\n",
    "        r_value = 1.0\n",
    "        g_value = 0.0\n",
    "        b_value = 1.0\n",
    "    if color == 'Yellow':\n",
    "        r_value = 1.0\n",
    "        g_value = 0.9\n",
    "        b_value = 0.0\n",
    "\n",
    "    color_list = []\n",
    "\n",
    "    unzipped_anneal_list = unzip_list(anneal_list)\n",
    "    \n",
    "    if scaling_factor == '':\n",
    "        max_index = max(unzipped_anneal_list[2])\n",
    "    else:\n",
    "        max_index = float(scaling_factor)\n",
    "\n",
    "    scaled_list = [i/max_index for i in unzipped_anneal_list[2]]\n",
    "\n",
    "    for i in range(x_dimension*y_dimension):\n",
    "        if i < len(unzipped_anneal_list[2]):\n",
    "            color_list += [matplotlib.colors.to_hex((scaled_list[i]*r_value, scaled_list[i]*g_value, \n",
    "            scaled_list[i]*b_value))]\n",
    "        else:\n",
    "            color_list += [\"#000000\"]\n",
    "    return color_list, max_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "tags": [
     "outputPrepend"
    ]
   },
   "outputs": [],
   "source": [
    "def find_neighbors(ind):\n",
    "    '''returns a list of the indices of the neighbors of the index given'''\n",
    "    neighbors = []\n",
    "    if 0 <= ind <= x_dimension-1:\n",
    "        # top row (inc. corners)\n",
    "        if ind == 0:\n",
    "            # top left corner\n",
    "            neighbors = [num_hex-1, num_hex-x_dimension, x_dimension-1, 2*x_dimension-1, ind+1,                  ind+x_dimension]\n",
    "        elif ind == (x_dimension-1):\n",
    "            # top right corner\n",
    "            neighbors = [ind-1, ind+x_dimension, 0, ind+x_dimension-1, num_hex-2, num_hex-1]\n",
    "        else:\n",
    "            # non-corner top row\n",
    "            neighbors = [ind-1, ind+1, ind+x_dimension, ind+num_hex-x_dimension-1, \n",
    "            ind+num_hex-x_dimension, ind+x_dimension-1]\n",
    "\n",
    "    elif (num_hex - x_dimension) <= ind <= num_hex -1:\n",
    "        if ind == (num_hex-x_dimension):\n",
    "            # bottom left corner\n",
    "            neighbors = [ind+1, ind-x_dimension, ind-x_dimension+1, 0, 1, num_hex-1]\n",
    "        elif ind == (num_hex-1):\n",
    "            # bottom right corner\n",
    "            neighbors = [ind-1, ind-x_dimension, ind-x_dimension+1, 0, x_dimension-1,\n",
    "            num_hex-2*x_dimension]\n",
    "        else:\n",
    "            # non-corner bottom row\n",
    "            neighbors = [ind-1, ind+1, ind-x_dimension, ind-x_dimension+1, ind-num_hex+x_dimension,\n",
    "            ind-num_hex+x_dimension+1]\n",
    "    elif ind % y_dimension == 0 and (ind/y_dimension)%2 == 1:\n",
    "        # \"inner\" left edge (not top or bottom row)\n",
    "        neighbors = [ind+x_dimension-1, ind+1, ind-x_dimension, ind-x_dimension+1, ind+x_dimension, \n",
    "        ind+x_dimension+1]\n",
    "    elif ind % y_dimension == 0 and (ind/y_dimension)%2 == 0:\n",
    "        # \"outer\" left edge (not top or bottom row)\n",
    "        neighbors = [ind-1, ind+1, ind+x_dimension, ind+2*x_dimension-1, ind-x_dimension, \n",
    "        ind+x_dimension-1]\n",
    "    elif (ind+1) % y_dimension == 0 and ((ind+1)/y_dimension)%2 == 0:\n",
    "        # \"outer\" right edge (not top or bottom row)\n",
    "        neighbors = [ind-1, ind+1, ind-x_dimension, ind-x_dimension+1, ind+x_dimension, \n",
    "        ind-2*x_dimension+1]\n",
    "    elif (ind+1) % y_dimension == 0 and ((ind+1)/y_dimension)%2 == 1:\n",
    "        # \"inner\" right edge (not top or bottom row)\n",
    "        neighbors = [ind-1, ind-x_dimension-1, ind-x_dimension, ind-x_dimension+1, ind+x_dimension, \n",
    "        ind+x_dimension-1]\n",
    "    else:\n",
    "        # middle\n",
    "        neighbors = [ind-1, ind+1, ind-x_dimension, ind-x_dimension+1, ind+x_dimension, \n",
    "        ind+x_dimension+1]\n",
    "    return neighbors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "# initially find fitness\n",
    "def find_fitness(anneal_list):\n",
    "    fitness = 0\n",
    "    for i in range(len(anneal_list)):\n",
    "        neighbors = find_neighbors(i)\n",
    "        sum_neighbor_score = 0\n",
    "        for index in neighbors:\n",
    "            intersection = [value for value in anneal_list[index][1] if value in anneal_list[i][1]]\n",
    "            if len(anneal_list[index][1]) + len(anneal_list[i][1]) != 0:\n",
    "                jaccard = len(intersection)/(len(anneal_list[index][1]) + len(anneal_list[i][1]))\n",
    "            else:\n",
    "                jaccard = 0.0\n",
    "            sum_neighbor_score += jaccard\n",
    "        hex_list = list(anneal_list[i])\n",
    "        hex_list[3] = sum_neighbor_score\n",
    "        hex_tuple = tuple(hex_list)\n",
    "        anneal_list[i] = hex_tuple\n",
    "        fitness += sum_neighbor_score\n",
    "    return fitness, anneal_list\n",
    "\n",
    "# take indices of swapped hexagons\n",
    "def find_swapped_fitness(anneal_list, swapped_a, swapped_b, old_fitness):\n",
    "    neighbors_a = find_neighbors(swapped_a)\n",
    "    neighbors_b = find_neighbors(swapped_b)\n",
    "    hexagons_to_update = [swapped_a, swapped_b] + neighbors_a + neighbors_b\n",
    "\n",
    "    new_fitness = 0\n",
    "    # Recalculate scores for all hexagons that need updating\n",
    "    for hex in hexagons_to_update:\n",
    "\n",
    "        # subtract out the swapped neighbor fitnesses because they are changing \n",
    "        old_fitness -= anneal_list[hex][3]\n",
    "\n",
    "        neighbors = find_neighbors(hex)\n",
    "        sum_neighbor_score = 0\n",
    "        for index in neighbors:\n",
    "            intersection = [value for value in anneal_list[index][1] if value in anneal_list[hex][1]]\n",
    "            if len(anneal_list[index][1]) + len(anneal_list[hex][1]) != 0:\n",
    "                jaccard = len(intersection)/(len(anneal_list[index][1]) + len(anneal_list[hex][1]))\n",
    "            else:\n",
    "                jaccard = 0.0\n",
    "            sum_neighbor_score += jaccard\n",
    "        hex_list = list(anneal_list[hex])\n",
    "        hex_list[3] = sum_neighbor_score\n",
    "        hex_tuple = tuple(hex_list)\n",
    "        anneal_list[hex] = hex_tuple\n",
    "        new_fitness += sum_neighbor_score\n",
    "    return old_fitness + new_fitness, anneal_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "def unzip_list(anneal_list):\n",
    "    unzipped_list = zip(*anneal_list)\n",
    "    return list(unzipped_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annealing(anneal_list, steps, old_fitness):\n",
    "    # returns unzipped list\n",
    "    for i in range(steps):\n",
    "        index_a = randint(0, num_hex-1)\n",
    "        index_b = randint(0, num_hex-1)\n",
    "        anneal_list[index_a], anneal_list[index_b] = anneal_list[index_b], anneal_list[index_a]\n",
    "        new_fitness, anneal_list = find_swapped_fitness(anneal_list, index_a, index_b, old_fitness)\n",
    "        if new_fitness <= old_fitness:\n",
    "            # swap back\n",
    "            anneal_list[index_a], anneal_list[index_b] = anneal_list[index_b], anneal_list[index_a]\n",
    "        else:\n",
    "            # finalize the swap by resetting old_fitness\n",
    "            old_fitness = new_fitness\n",
    "    return anneal_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def init_chart():\n",
    "  chart_id = 'mychart-' + str(uuid.uuid4())\n",
    "  display(HTML('<script src=\"/static/components/requirejs/require.js\"></script>'))\n",
    "  display(HTML(Template(dedent('''\n",
    "  <script>\n",
    "  require.config({\n",
    "    paths: {\n",
    "      'd3': 'https://cdnjs.cloudflare.com/ajax/libs/d3/5.16.0/d3.min',\n",
    "      'd3-hexbin': 'https://d3js.org/d3-hexbin.v0.2.min',\n",
    "    },\n",
    "    shim: {\n",
    "      'd3-hexbin': ['d3']\n",
    "    }\n",
    "  })\n",
    "\n",
    "  // If we configure mychart via url, we can eliminate this define here\n",
    "  define($chart_id, ['d3', 'd3-hexbin'], function(d3, d3_hexbin) {\n",
    "    return function (figure_id, numA, numB, colorList, libraryList, indices) {\n",
    "      var margin = {top: 50, right: 20, bottom: 20, left: 50},\n",
    "        width = 850 - margin.left - margin.right,\n",
    "        height = 350 - margin.top - margin.bottom;\n",
    "\n",
    "      // append the svg object to the body of the page\n",
    "      var svG = d3.select('#' + figure_id)\n",
    "                  .attr(\"width\", width + margin.left + margin.right)\n",
    "                  .attr(\"height\", height + margin.top + margin.bottom)\n",
    "                  .append(\"g\")\n",
    "                  .attr(\"transform\", \"translate(\" + margin.left + \",\" + margin.top + \")\");\n",
    "      \n",
    "      //The number of columns and rows of the heatmap\n",
    "      var MapColumns = numA,\n",
    "          MapRows = numB;\n",
    "\n",
    "      //The maximum radius the hexagons can have to still fit the screen\n",
    "      var hexRadius = d3.min([width/((MapColumns + 0.5) * Math.sqrt(3)), height/((MapRows + 1/3) * 1.5)]);\n",
    "\n",
    "      //Calculate the center position of each hexagon\n",
    "      var points = [];\n",
    "      for (var i = 0; i < MapRows; i++) {\n",
    "          for (var j = 0; j < MapColumns; j++) {\n",
    "              var x = hexRadius * j * Math.sqrt(3)\n",
    "              //Offset each uneven row by half of a \"hex-width\" to the right\n",
    "              if(i%2 === 1) x += (hexRadius * Math.sqrt(3))/2\n",
    "              var y = hexRadius * i * 1.5\n",
    "              points.push([x,y])\n",
    "          }\n",
    "      }\n",
    "\n",
    "      //Set the hexagon radius\n",
    "      var hexbin = d3_hexbin.hexbin().radius(hexRadius);\n",
    "\n",
    "      svG.append(\"g\")\n",
    "        .selectAll(\".hexagon\")\n",
    "        .data(hexbin(points))\n",
    "        .enter().append(\"path\")\n",
    "        .attr(\"class\", \"hexagon\")\n",
    "        .attr(\"d\", function (d) {\n",
    "            return \"M\" + d.x + \",\" + d.y + hexbin.hexagon();\n",
    "        })\n",
    "        .attr(\"stroke\", \"white\")\n",
    "        .attr(\"stroke-width\", \"1px\")\n",
    "        .style(\"fill\", function (d,i) { return colorList[i]; })\n",
    "        .on(\"mouseover\", mover)\n",
    "        .on(\"mouseout\", mout)\n",
    "        .append(\"svg:title\")\n",
    "        .text(function(d,i) { return libraryList[i].concat(\" \").concat(indices[i]); });\n",
    "\n",
    "      // Mouseover function\n",
    "      function mover(d) {\n",
    "      d3.select(this)\n",
    "        .transition().duration(10)  \n",
    "        .style(\"fill-opacity\", 0.3)\n",
    "      };\n",
    "\n",
    "      // Mouseout function\n",
    "      function mout(d) { \n",
    "      d3.select(this)\n",
    "        .transition().duration(10)\n",
    "        .style(\"fill-opacity\", 1)\n",
    "      };\n",
    "\n",
    "  }\n",
    "\n",
    "  })\n",
    "  </script>\n",
    "  ''')).substitute({ 'chart_id': repr(chart_id) })))\n",
    "  return chart_id\n",
    "\n",
    "def Canvas(numA, numB, colorList, libraryList, indices):\n",
    "  chart_id = init_chart()\n",
    "  display(HTML(Template(dedent('''\n",
    "  <svg id=$figure_id></svg>\n",
    "  <script>\n",
    "  require([$chart_id], function(mychart) {\n",
    "    mychart($figure_id, $numA, $numB, $colorList, $libraryList, $indices)\n",
    "  })\n",
    "  </script>\n",
    "  ''')).substitute({\n",
    "      'chart_id': repr(chart_id),\n",
    "      'figure_id': repr('fig-' + str(uuid.uuid4())),\n",
    "      'numA': repr(numA),\n",
    "      'numB': repr(numB),\n",
    "      'colorList': repr(color_list),\n",
    "      'libraryList': repr(list(unzipped_anneal_list[0])),\n",
    "      'indices': repr(list(unzipped_anneal_list[2]))\n",
    "  })))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/plain": "<IPython.core.display.HTML object>",
      "text/html": "<script src=\"/static/components/requirejs/require.js\"></script>"
     },
     "metadata": {}
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": "<IPython.core.display.HTML object>",
      "text/html": "\n<script>\nrequire.config({\n  paths: {\n    'd3': 'https://cdnjs.cloudflare.com/ajax/libs/d3/5.16.0/d3.min',\n    'd3-hexbin': 'https://d3js.org/d3-hexbin.v0.2.min',\n  },\n  shim: {\n    'd3-hexbin': ['d3']\n  }\n})\n\n// If we configure mychart via url, we can eliminate this define here\ndefine('mychart-03c4d531-8428-4420-8a4b-32ff745793f0', ['d3', 'd3-hexbin'], function(d3, d3_hexbin) {\n  return function (figure_id, numA, numB, colorList, libraryList, indices) {\n    var margin = {top: 50, right: 20, bottom: 20, left: 50},\n      width = 850 - margin.left - margin.right,\n      height = 350 - margin.top - margin.bottom;\n\n    // append the svg object to the body of the page\n    var svG = d3.select('#' + figure_id)\n                .attr(\"width\", width + margin.left + margin.right)\n                .attr(\"height\", height + margin.top + margin.bottom)\n                .append(\"g\")\n                .attr(\"transform\", \"translate(\" + margin.left + \",\" + margin.top + \")\");\n\n    //The number of columns and rows of the heatmap\n    var MapColumns = numA,\n        MapRows = numB;\n\n    //The maximum radius the hexagons can have to still fit the screen\n    var hexRadius = d3.min([width/((MapColumns + 0.5) * Math.sqrt(3)), height/((MapRows + 1/3) * 1.5)]);\n\n    //Calculate the center position of each hexagon\n    var points = [];\n    for (var i = 0; i < MapRows; i++) {\n        for (var j = 0; j < MapColumns; j++) {\n            var x = hexRadius * j * Math.sqrt(3)\n            //Offset each uneven row by half of a \"hex-width\" to the right\n            if(i%2 === 1) x += (hexRadius * Math.sqrt(3))/2\n            var y = hexRadius * i * 1.5\n            points.push([x,y])\n        }\n    }\n\n    //Set the hexagon radius\n    var hexbin = d3_hexbin.hexbin().radius(hexRadius);\n\n    svG.append(\"g\")\n      .selectAll(\".hexagon\")\n      .data(hexbin(points))\n      .enter().append(\"path\")\n      .attr(\"class\", \"hexagon\")\n      .attr(\"d\", function (d) {\n          return \"M\" + d.x + \",\" + d.y + hexbin.hexagon();\n      })\n      .attr(\"stroke\", \"white\")\n      .attr(\"stroke-width\", \"1px\")\n      .style(\"fill\", function (d,i) { return colorList[i]; })\n      .on(\"mouseover\", mover)\n      .on(\"mouseout\", mout)\n      .append(\"svg:title\")\n      .text(function(d,i) { return libraryList[i].concat(\" \").concat(indices[i]); });\n\n    // Mouseover function\n    function mover(d) {\n    d3.select(this)\n      .transition().duration(10)  \n      .style(\"fill-opacity\", 0.3)\n    };\n\n    // Mouseout function\n    function mout(d) { \n    d3.select(this)\n      .transition().duration(10)\n      .style(\"fill-opacity\", 1)\n    };\n\n}\n\n})\n</script>\n"
     },
     "metadata": {}
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": "<IPython.core.display.HTML object>",
      "text/html": "\n<svg id='fig-0e6c2a81-3e37-4826-8169-2f825147a085'></svg>\n<script>\nrequire(['mychart-03c4d531-8428-4420-8a4b-32ff745793f0'], function(mychart) {\n  mychart('fig-0e6c2a81-3e37-4826-8169-2f825147a085', 22, 22, ['#000000', '#000000', '#000000', '#a100a1', '#000000', '#390039', '#000000', '#000000', '#8d008d', '#000000', '#000000', '#330033', '#000000', '#000000', '#ff00ff', '#000000', '#3a003a', '#330033', '#3c003c', '#5e005e', '#000000', '#380038', '#330033', '#000000', '#340034', '#000000', '#350035', '#000000', '#000000', '#000000', '#360036', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#000000', '#3d003d', '#000000', '#390039', '#000000', '#000000', '#5b005b', '#5a005a', '#5b005b', '#000000', '#3b003b', '#000000', '#000000', '#000000', '#380038', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3e003e', '#000000', '#000000', '#000000', '#c900c9', '#7c007c', '#3d003d', '#000000', '#390039', '#3c003c', '#370037', '#000000', '#000000', '#000000', '#000000', '#330033', '#000000', '#000000', '#330033', '#000000', '#000000', '#620062', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3e003e', '#3d003d', '#000000', '#000000', '#000000', '#000000', '#000000', '#850085', '#000000', '#3d003d', '#910091', '#000000', '#000000', '#2e002e', '#000000', '#000000', '#950095', '#360036', '#2e002e', '#000000', '#3a003a', '#710071', '#860086', '#330033', '#000000', '#000000', '#3c003c', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#3c003c', '#000000', '#3a003a', '#000000', '#000000', '#3a003a', '#000000', '#000000', '#390039', '#000000', '#000000', '#af00af', '#7d007d', '#3e003e', '#360036', '#760076', '#000000', '#3a003a', '#3a003a', '#000000', '#000000', '#000000', '#330033', '#000000', '#330033', '#000000', '#3d003d', '#3b003b', '#000000', '#000000', '#000000', '#630063', '#000000', '#a300a3', '#000000', '#000000', '#3e003e', '#3b003b', '#000000', '#3c003c', '#730073', '#390039', '#000000', '#000000', '#000000', '#330033', '#000000', '#3d003d', '#710071', '#370037', '#000000', '#3e003e', '#000000', '#000000', '#000000', '#900090', '#000000', '#360036', '#000000', '#650065', '#360036', '#000000', '#000000', '#000000', '#000000', '#000000', '#390039', '#380038', '#370037', '#390039', '#390039', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#b200b2', '#3e003e', '#3e003e', '#2d002d', '#000000', '#320032', '#000000', '#6f006f', '#000000', '#390039', '#760076', '#000000', '#3d003d', '#3d003d', '#370037', '#000000', '#3e003e', '#350035', '#3b003b', '#000000', '#000000', '#650065', '#000000', '#000000', '#990099', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#000000', '#000000', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#5b005b', '#000000', '#390039', '#000000', '#aa00aa', '#000000', '#3c003c', '#b900b9', '#7c007c', '#000000', '#000000', '#000000', '#000000', '#b500b5', '#790079', '#a500a5', '#3e003e', '#000000', '#000000', '#5a005a', '#000000', '#000000', '#7c007c', '#3e003e', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#000000', '#3c003c', '#000000', '#3c003c', '#000000', '#000000', '#000000', '#000000', '#5a005a', '#000000', '#000000', '#890089', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#000000', '#360036', '#750075', '#6f006f', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3a003a', '#6d006d', '#000000', '#000000', '#000000', '#000000', '#a600a6', '#000000', '#770077', '#6b006b', '#000000', '#330033', '#350035', '#3a003a', '#000000', '#360036', '#970097', '#9b009b', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#3b003b', '#000000', '#000000', '#000000', '#000000', '#000000', '#6a006a', '#000000', '#350035', '#000000', '#6d006d', '#000000', '#340034', '#000000', '#000000', '#000000', '#370037', '#000000', '#800080', '#2d002d', '#000000', '#390039', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3e003e', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#390039', '#9b009b', '#000000', '#000000', '#790079', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#ab00ab', '#000000', '#3e003e', '#6b006b', '#000000', '#380038', '#860086', '#000000', '#000000', '#000000', '#000000', '#380038', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#7a007a', '#000000', '#000000', '#000000', '#2f002f', '#000000', '#330033', '#000000', '#000000', '#000000', '#000000', '#3a003a', '#000000', '#3e003e', '#000000', '#3c003c', '#940094', '#000000', '#000000', '#000000', '#000000', '#000000', '#3a003a', '#3e003e', '#000000', '#3c003c', '#000000', '#380038', '#000000', '#7a007a', '#710071', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#000000', '#9c009c', '#000000', '#340034', '#390039', '#000000', '#7a007a', '#700070', '#000000', '#000000', '#780078', '#3a003a', '#000000', '#000000', '#320032', '#610061', '#3a003a', '#3a003a', '#000000', '#000000', '#5f005f', '#000000', '#000000', '#000000', '#000000', '#3c003c', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#000000', '#670067', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000'], ['4-hydroxytamoxifen, Dexamethasone, and Retinoic Acids Regulation of p27 Expression WP3879', 'ACE Inhibitor Pathway WP554', 'AGE/RAGE pathway WP2324', 'AMP-activated Protein Kinase (AMPK) Signaling WP1403', 'ATM Signaling Network in Development and Disease  WP3878', 'ATM Signaling Pathway WP2516', 'ATR Signaling WP3875', 'Acetylcholine Synthesis WP528', 'Adipogenesis WP236', 'Aflatoxin B1 metabolism WP699', 'Alanine and aspartate metabolism WP106', 'Allograft Rejection WP2328', 'Alpha 6 Beta 4 signaling pathway WP244', 'Alzheimers Disease WP2059', 'Amino Acid metabolism WP3925', 'Amplification and Expansion of Oncogenic Pathways as Metastatic Traits WP3678', 'Amyotrophic lateral sclerosis (ALS) WP2447', 'Androgen receptor signaling pathway WP138', 'Angiogenesis WP1539', 'Angiopoietin Like Protein 8 Regulatory Pathway WP3915', 'ApoE and miR-146 in inflammation and atherosclerosis WP3926', 'Apoptosis-related network due to altered Notch3 in ovarian cancer WP2864', 'Apoptosis Modulation and Signaling WP1772', 'Apoptosis Modulation by HSP70 WP384', 'Apoptosis WP254', 'Arachidonate Epoxygenase / Epoxide Hydrolase WP678', 'Arrhythmogenic Right Ventricular Cardiomyopathy WP2118', 'Aryl Hydrocarbon Receptor Pathway WP2873', 'Aryl Hydrocarbon Receptor WP2586', 'Arylamine metabolism WP694', 'Association Between Physico-Chemical Features and Toxicity Associated Pathways WP3680', 'BDNF-TrkB Signaling WP3676', 'BMP2-WNT4-FOXO1 Pathway in Human Primary Endometrial Stromal Cell Differentiation WP3876', 'BMP Signaling Pathway in Eyelid Development WP3927', 'B Cell Receptor Signaling Pathway WP23', 'Benzene metabolism WP3891', 'Benzo(a)pyrene metabolism WP696', 'Bile Acids synthesis and enterohepatic circulation  WP4389', 'Biogenic Amine Synthesis WP550', 'Bladder Cancer WP2828', 'Blood Clotting Cascade WP272', 'Bone Morphogenic Protein (BMP) Signalling and Regulation WP1425', 'Brain-Derived Neurotrophic Factor (BDNF) signaling pathway WP2380', 'Breast cancer pathway WP4262', 'Calcium Regulation in the Cardiac Cell WP536', 'Caloric restriction and aging WP4191', 'Cannabinoid receptor signaling WP3869', 'Canonical  and Non-canonical Notch signaling WP3845', 'Canonical and Non-Canonical TGF-B signaling WP3874', 'Cardiac Hypertrophic Response WP2795', 'Cardiac Progenitor Differentiation WP2406', 'Catalytic cycle of mammalian Flavin-containing MonoOxygenases (FMOs) WP688', 'Cell-type Dependent Selectivity of CCK2R Signaling WP3679', 'Cell Cycle WP179', 'Cell Differentiation - Index WP2029', 'Cell Differentiation - Index expanded WP2023', 'Cells and Molecules involved in local acute inflammatory response  WP4493', 'Cerebral Organic Acidurias, including diseases WP4519', 'Chemokine signaling pathway WP3929', 'Cholesterol Biosynthesis Pathway WP197', 'Chromosomal and microsatellite instability in colorectal cancer  WP4216', 'Ciliary landscape WP4352', 'Circadian rhythm related genes WP3594', 'Classical pathway of steroidogenesis, including diseases WP4523', 'Codeine and Morphine Metabolism WP1604', 'Common Pathways Underlying Drug Addiction WP2636', 'Complement Activation WP545', 'Complement and Coagulation Cascades WP558', 'Composition of Lipid Particles WP3601', 'Constitutive Androstane Receptor Pathway WP2875', 'Copper homeostasis WP3286', 'Cori Cycle WP1946', 'Corticotropin-releasing hormone signaling pathway WP2355', 'Cysteine and methionine catabolism WP4504', 'Cytokines and Inflammatory Response WP530', 'Cytoplasmic Ribosomal Proteins WP477', 'Cytosine methylation WP3585', 'DDX1 as a regulatory component of the Drosha microprocessor WP2942', 'DNA Damage Response (only ATM dependent) WP710', 'DNA Damage Response WP707', 'DNA IR-Double Strand Breaks (DSBs) and cellular response via ATM WP3959', 'DNA IR-damage and cellular response via ATR WP4016', 'DNA Mismatch Repair WP531', 'DNA Replication WP466', 'Degradation pathway of sphingolipids, including diseases WP4153', 'Deregulation of Rab and Rab Effector Genes in Bladder Cancer WP2291', 'Development and heterogeneity of the ILC family WP3893', 'Development of pulmonary dendritic cells and macrophage subsets WP3892', 'Differentiation Pathway WP2848', 'Differentiation of white and brown adipocyte    WP2895', 'Disorders of Folate Metabolism and Transport WP4259', 'Disorders of the Krebs cycle WP4236', 'Dopamine metabolism WP2436', 'Dopaminergic Neurogenesis WP2855', 'Drug Induction of Bile Acid Pathway WP2289', 'Dual hijack model of Vif in HIV infection WP3300', 'EBV LMP1 signaling WP262', 'EDA Signalling in Hair Follicle Development WP3930', 'EGF/EGFR Signaling Pathway WP437', 'EPO Receptor Signaling WP581', \"ERK Pathway in Huntington's Disease WP3853\", 'ESC Pluripotency Pathways WP3931', 'EV release from cardiac cells and their functional effects WP3297', 'Ebola Virus Pathway on Host WP4217', 'Ectoderm Differentiation WP2858', 'Effects of Nitric Oxide WP1995', 'Eicosanoid Synthesis WP167', 'Electron Transport Chain (OXPHOS system in mitochondria) WP111', 'Endochondral Ossification WP474', 'Endoderm Differentiation WP2853', 'Endometrial cancer WP4155', 'Endothelin Pathways WP2197', 'Energy Metabolism WP1541', 'Epithelial to mesenchymal transition in colorectal cancer WP4239', 'ErbB Signaling Pathway WP673', 'Estrogen Receptor Pathway WP2881', 'Estrogen metabolism WP697', 'Estrogen signaling pathway WP712', 'Ethanol effects on histone modifications WP3996', 'Ethanol metabolism resulting in production of ROS by CYP2E1 WP4269', 'Eukaryotic Transcription Initiation WP405', 'Exercise-induced Circadian Regulation WP410', 'Extracellular vesicle-mediated signaling in recipient cells WP2870', 'Extracellular vesicles in the crosstalk of cardiac cells WP4300', 'FTO Obesity Variant Mechanism WP3407', 'Factors and pathways affecting insulin-like growth factor (IGF1)-Akt signaling WP3850', 'Farnesoid X Receptor  Pathway WP2879', 'Fas Ligand (FasL) pathway and Stress induction of Heat Shock Proteins (HSP) regulation WP314', 'Fatty Acid Beta Oxidation WP143', 'Fatty Acid Biosynthesis WP357', 'Fatty Acid Omega Oxidation WP206', 'Ferroptosis WP4313', 'Fibrin Complement Receptor 3 Signaling Pathway WP4136', 'Fluoropyrimidine Activity WP1601', 'Focal Adhesion-PI3K-Akt-mTOR-signaling pathway WP3932', 'Focal Adhesion WP306', 'Folate-Alcohol and Cancer Pathway Hypotheses WP1589', 'Folate Metabolism WP176', 'Follicle Stimulating Hormone (FSH) signaling pathway WP2035', 'G1 to S cell cycle control WP45', 'G13 Signaling Pathway WP524', 'GABA receptor Signaling WP4159', 'GPCRs, Class A Rhodopsin-like WP455', 'GPCRs, Class B Secretin-like WP334', 'GPCRs, Class C Metabotropic glutamate, pheromone WP501', 'GPCRs, Other WP117', 'GPR40 Pathway WP3958', 'G Protein Signaling Pathways WP35', 'Gamma-Glutamyl Cycle for the biosynthesis and degradation of glutathione, including diseases WP4518', 'Ganglio Sphingolipid Metabolism WP1423', 'Gastric Cancer Network 1 WP2361', 'Gastric Cancer Network 2 WP2363', 'Gastric acid production WP2596', 'Gene regulatory network modelling somitogenesis  WP2854', 'Genes related to primary cilium development (based on CRISPR) WP4536', 'Genes targeted by miRNAs in adipocytes WP1992', 'Genotoxicity pathway WP4286', 'Glial Cell Differentiation WP2276', 'Globo Sphingolipid Metabolism WP1424', 'Glucocorticoid and Mineralcorticoid Metabolism WP237', 'Glucuronidation WP698', 'Glutathione metabolism WP100', 'Glycerophospholipid Biosynthetic Pathway WP2533', 'Glycogen Synthesis and Degradation WP500', 'Glycolysis and Gluconeogenesis WP534', 'Glycosylation and related congenital defects. WP4521', 'H19 action Rb-E2F1 signaling and CDK-Beta-catenin activity WP3969', 'HIF1A and PPARG regulation of glycolysis WP2456', 'Hair Follicle Development: Cytodifferentiation (Part 3 of 3) WP2840', 'Heart Development WP1591', 'Hedgehog Signaling Pathway WP47', 'Hedgehog Signaling Pathway WP4249', 'Hematopoietic Stem Cell Differentiation WP2849', 'Hematopoietic Stem Cell Gene Regulation by GABP alpha/beta Complex WP3657', 'Heme Biosynthesis WP561', 'Hepatitis C and Hepatocellular Carcinoma WP3646', 'Hereditary leiomyomatosis and renal cell carcinoma pathway WP4206', 'Hfe effect on hepcidin production WP3924', 'Hippo-Merlin Signaling Dysregulation WP4541', 'Hippo-Yap signaling pathway WP4537', 'Histone Modifications WP2369', 'Homologous recombination WP186', 'Human Complement System WP2806', 'Human Thyroid Stimulating Hormone (TSH) signaling pathway WP2032', 'Hypertrophy Model WP516', 'Hypothesized Pathways in Pathogenesis of Cardiovascular Disease WP3668', 'Hypothetical Craniofacial Development Pathway WP3655', 'ID signaling pathway WP53', 'IL-1 signaling pathway WP195', 'IL-2 Signaling Pathway WP49', 'IL-3 Signaling Pathway WP286', 'IL-4 Signaling Pathway WP395', 'IL-5 Signaling Pathway WP127', 'IL-6 signaling pathway WP364', 'IL-7 Signaling Pathway WP205', 'IL-9 Signaling Pathway WP22', 'IL-10 Anti-inflammatory Signaling Pathway  WP4495', 'IL1 and megakaryocytes in obesity WP2865', 'IL17 signaling pathway WP2112', 'Imatinib and Chronic Myeloid Leukemia WP3640', 'Inflammatory Response Pathway WP453', 'Inhibition of exosome biogenesis and secretion by Manumycin A in CRPC cells WP4301', 'Initiation of transcription and translation elongation at the HIV-1 LTR WP3414', 'Insulin Signaling WP481', 'Insulin signalling in human adipocytes (diabetic condition) WP3635', 'Insulin signalling in human adipocytes (normal condition) WP3634', 'Integrated Breast Cancer Pathway WP1984', 'Integrated Cancer Pathway WP1971', 'Integrin-mediated Cell Adhesion WP185', 'Interactome of polycomb repressive complex 2 (PRC2)  WP2916', 'Interferon type I signaling pathways WP585', 'Interleukin-1 Induced Activation of NF-kappa-B WP3656', 'Interleukin-11 Signaling Pathway WP2332', 'Intraflagellar transport proteins binding to dynein WP4532', 'Irinotecan Pathway WP229', 'Iron metabolism in placenta WP2007', 'Kennedy pathway from Sphingolipids WP3933', 'Kit receptor signaling pathway WP304', 'Leptin Insulin Overlap WP3935', 'Leptin and adiponectin WP3934', 'Leptin signaling pathway WP2034', 'Lipid Metabolism Pathway WP3965', 'Liver X Receptor Pathway WP2874', 'LncRNA-mediated mechanisms of therapeutic resistance WP3672', 'LncRNA involvement in canonical Wnt signaling and colorectal cancer WP4258', 'Lung fibrosis WP3624', 'MAPK Cascade WP422', 'MAPK Signaling Pathway WP382', 'MAPK  and NFkB Signalling Pathways Inhibited by Yersinia YopJ WP3849', 'MECP2 and Associated Rett Syndrome WP3584', 'MET in type 1 papillary renal cell carcinoma WP4205', 'MFAP5-mediated ovarian cancer cell motility and invasiveness WP3301', 'MTHFR deficiency WP4288', 'Macrophage markers WP4146', 'Mammary gland development pathway - Embryonic development (Stage 1 of 4) WP2813', 'Mammary gland development pathway - Involution (Stage 4 of 4) WP2815', 'Mammary gland development pathway - Pregnancy and lactation (Stage 3 of 4) WP2817', 'Mammary gland development pathway - Puberty (Stage 2 of 4) WP2814', 'Matrix Metalloproteinases WP129', 'Mechanoregulation and pathology of YAP/TAZ via Hippo and non-Hippo mechanisms WP4534', 'Melatonin metabolism and effects WP3298', 'Mesodermal Commitment Pathway WP2857', 'Metabolic pathway of LDL, HDL and TG, including diseases WP4522', 'Metabolic reprogramming in colon cancer WP4290', 'Metabolism of Spingolipids in ER and Golgi apparatus WP4142', 'Metapathway biotransformation Phase I and II WP702', 'Metastatic brain tumor WP2249', 'Methionine De Novo and Salvage Pathway WP3580', 'Methionine metabolism leading to Sulphur Amino Acids and related disorders WP4292', 'Methylation Pathways WP704', 'Mevalonate pathway WP3963', 'MicroRNA for Targeting Cancer Growth and Vascularization in Glioblastoma WP3593', 'MicroRNAs in cardiomyocyte hypertrophy WP1544', 'Microglia Pathogen Phagocytosis Pathway WP3937', 'Mitochondrial Gene Expression WP391', 'Mitochondrial LC-Fatty Acid Beta-Oxidation WP368', 'Mitochondrial complex I assembly model OXPHOS system WP4324', 'Molybdenum cofactor (Moco) biosynthesis WP4507', 'Monoamine GPCRs WP58', 'Monoamine Transport WP727', 'Myometrial Relaxation and Contraction Pathways WP289', 'NAD+ biosynthetic pathways WP3645', 'NAD+ metabolism WP3644', 'NAD Biosynthesis II (from tryptophan) WP2485', 'NAD metabolism, sirtuins and aging WP3630', 'NLR Proteins WP288', 'NO/cGMP/PKG mediated Neuroprotection WP4008', 'NOTCH1 regulation of human endothelial cell calcification WP3413', 'NRF2-ARE regulation WP4357', 'NRF2 pathway WP2884', 'Nanomaterial induced apoptosis WP2507', 'Nanoparticle-mediated activation of receptor signaling WP2643', 'Nanoparticle triggered autophagic cell death WP2509', 'Nanoparticle triggered regulated necrosis WP2513', 'Neovascularisation processes WP4331', 'Neural Crest Differentiation WP2064', 'Nicotine Activity on Dopaminergic Neurons WP1602', 'Nicotine Metabolism WP1600', 'Non-genomic actions of 1,25 dihydroxyvitamin D3 WP4341', 'Non-homologous end joining WP438', 'Non-small cell lung cancer WP4255', 'Nonalcoholic fatty liver disease WP4396', 'Notch Signaling Pathway WP61', 'Notch Signaling WP268', 'Nuclear Receptors Meta-Pathway WP2882', 'Nuclear Receptors WP170', 'Nuclear Receptors in Lipid Metabolism and Toxicity WP299', 'Nucleotide-binding Oligomerization Domain (NOD) pathway WP1433', 'Nucleotide GPCRs WP80', 'Nucleotide Metabolism WP404', 'Oligodendrocyte Specification and differentiation(including remyelination), leading to Myelin Components for CNS WP4304', 'Oncostatin M Signaling Pathway WP2374', 'One Carbon Metabolism WP241', 'One carbon metabolism and related pathways WP3940', 'Osteoblast Signaling WP322', 'Osteoclast Signaling WP12', 'Osteopontin Signaling WP1434', 'Ovarian Infertility Genes WP34', 'Overview of nanoparticle effects WP3287', 'Oxidation by Cytochrome P450 WP43', 'Oxidative Damage WP3941', 'Oxidative Stress WP408', 'Oxidative phosphorylation WP623', 'Oxysterols derived from cholesterol WP4545', 'PDGFR-beta pathway WP3972', 'PDGF Pathway WP2526', 'PI3K-AKT-mTOR signaling pathway and therapeutic opportunities WP3844', 'PI3K-Akt Signaling Pathway WP4172', 'PI3K/AKT/mTOR - VitD3 Signalling WP4141', 'PPAR Alpha Pathway WP2878', 'PPAR signaling pathway WP3942', 'PTF1A related regulatory pathway WP4147', 'Pancreatic adenocarcinoma pathway WP4263', 'Parkin-Ubiquitin Proteasomal System pathway WP2359', 'Parkinsons Disease Pathway WP2371', 'Pathogenic Escherichia coli infection WP2272', 'Pathways Affected in Adenoid Cystic Carcinoma WP3651', 'Pathways Regulating Hippo Signaling WP4540', 'Pathways in clear cell renal cell carcinoma WP4018', 'Pentose Phosphate Metabolism WP134', 'Peptide GPCRs WP24', 'Phase I biotransformations, non P450 WP136', 'Phosphodiesterases in neuronal function WP4222', 'Photodynamic therapy-induced AP-1 survival signaling. WP3611', 'Photodynamic therapy-induced HIF-1 survival signaling WP3614', 'Photodynamic therapy-induced NF-kB survival signaling WP3617', 'Photodynamic therapy-induced NFE2L2 (NRF2) survival signaling WP3612', 'Photodynamic therapy-induced unfolded protein response WP3613', 'Physiological and Pathological Hypertrophy  of the Heart WP1528', 'Phytochemical activity on NRF2 transcriptional activation WP3', 'Platelet-mediated interactions with vascular and circulating cells WP4462', 'Prader-Willi and Angelman Syndrome WP3998', 'Pregnane X Receptor pathway WP2876', 'Preimplantation Embryo WP3527', 'Primary Focal Segmental Glomerulosclerosis FSGS WP2572', 'Prion disease pathway WP3995', 'Prolactin Signaling Pathway WP2037', 'Prostaglandin Synthesis and Regulation WP98', 'Proteasome Degradation WP183', 'Purine metabolism WP4224', 'Pyrimidine metabolism WP4022', 'Pyrimidine metabolism and related diseases WP4225', 'RAC1/PAK1/p38/MMP2 Pathway WP3303', 'RANKL/RANK (Receptor activator of NFKB (ligand)) Signaling Pathway WP2018', 'RIG-I-like Receptor Signaling WP3865', 'RalA downstream regulated genes WP2290', 'Ras Signaling WP4223', 'Regulation of Actin Cytoskeleton WP51', 'Regulation of Apoptosis by Parathyroid Hormone-related Protein WP3872', 'Regulation of Microtubule Cytoskeleton WP2038', 'Regulation of Wnt/B-catenin Signaling by Small Molecule Compounds WP3664', 'Regulation of sister chromatid separation at the metaphase-anaphase transition WP4240', 'Regulation of toll-like receptor signaling pathway WP1449', 'Resistin as a regulator of inflammation WP4481', 'Retinoblastoma Gene in Cancer WP2446', 'Rett syndrome causing genes WP4312', 'Robo4 and VEGF Signaling Pathways Crosstalk WP3943', 'Role Altered Glycolysation of MUC1 in Tumour Microenvironment WP4480', 'Role of Osx and miRNAs in tooth development WP3971', 'SCFA and skeletal muscle substrate metabolism WP4030', 'SREBF and miR33 in cholesterol and lipid homeostasis WP2011', 'SRF and miRs in Smooth Muscle Differentiation and Proliferation WP1991', 'Secretion of Hydrochloric Acid in Parietal Cells WP2597', 'Selective expression of chemokine receptors during T-cell polarization WP4494', 'Selenium Metabolism and Selenoproteins WP28', 'Selenium Micronutrient Network WP15', 'Senescence and Autophagy in Cancer WP615', 'Serotonin Receptor 2 and ELK-SRF/GATA4 signaling WP732', 'Serotonin Receptor 4/6/7 and NR3C Signaling WP734', 'Serotonin Transporter Activity WP1455', 'Serotonin and anxiety-related events WP3944', 'Serotonin and anxiety WP3947', 'Signal Transduction of S1P Receptor WP26', 'Signal transduction through IL1R WP4496', 'Signaling Pathways in Glioblastoma WP2261', 'Signaling of Hepatocyte Growth Factor Receptor WP313', 'Simplified Depiction of MYD88 Distinct Input-Output Pathway WP3877', 'Simplified Interaction Map Between LOXL4 and Oxidative Stress Pathway WP3670', 'Sleep regulation WP3591', 'Small Ligand GPCRs WP247', 'Somatroph axis (GH) and its relationship to dietary restriction and aging WP4186', 'Sphingolipid pathway WP1422', 'Spinal Cord Injury WP2431', 'Splicing factor NOVA regulated synaptic proteins WP4148', 'Statin Pathway WP430', 'Steroid Biosynthesis WP496', 'Sterol Regulatory Element-Binding Proteins (SREBP) signalling WP1982', 'Striated Muscle Contraction Pathway WP383', 'Structural Pathway of Interleukin 1 (IL-1) WP2637', 'Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways WP706', 'Sulfation Biotransformation Reaction WP692', 'Sulindac Metabolic Pathway WP2542', 'Supression of HMGB1 mediated inflammation by THBD WP4479', 'Synaptic Vesicle Pathway WP2267', 'Synaptic signaling pathways associated with autism spectrum disorder WP4539', 'Synthesis and Degradation of Ketone Bodies WP311', 'T-Cell Receptor and Co-stimulatory Signaling WP2583', 'T-Cell antigen Receptor (TCR)  Signaling Pathway WP69', 'T-Cell antigen Receptor (TCR) pathway during Staphylococcus aureus infection WP3863', 'TCA Cycle (aka Krebs or citric acid cycle) WP78', 'TCA Cycle Nutrient Utilization and Invasiveness of Ovarian Cancer WP2868', 'TCA Cycle and Deficiency of Pyruvate Dehydrogenase complex (PDHc) WP2453', 'TFs Regulate miRNAs related to cardiac hypertrophy WP1559', 'TGF-B Signaling in Thyroid Cells for Epithelial-Mesenchymal Transition WP3859', 'TGF-beta Receptor Signaling WP560', 'TGF-beta Signaling Pathway WP366', 'TLR4 Signaling and Tolerance WP3851', 'TNF alpha Signaling Pathway WP231', 'TNF related weak inducer of apoptosis (TWEAK) Signaling Pathway WP2036', 'TP53 Network WP1742', 'TYROBP Causal Network WP3945', 'Tamoxifen metabolism WP691', 'Target Of Rapamycin (TOR) Signaling WP1471', 'Tgif disruption of Shh signaling WP3674', 'The alternative pathway of fetal androgen synthesis WP4524', 'The effect of progerin on the involved genes in Hutchinson-Gilford Progeria Syndrome WP4320', 'The human immune response to tuberculosis WP4197', 'Thermogenesis WP4321', 'Thiamine metabolic pathways WP4297', 'Thymic Stromal LymphoPoietin (TSLP) Signaling Pathway WP2203', 'Thyroxine (Thyroid Hormone) Production WP1981', 'Toll-like Receptor Signaling Pathway WP75', 'Toll-like Receptor Signaling WP3858', 'Trans-sulfuration and one carbon metabolism WP2525', 'Trans-sulfuration pathway WP2333', 'Transcription co-factors SKI and SKIL protein partners WP4533', 'Transcription factor regulation in adipogenesis WP3599', 'Transcriptional cascade regulating adipogenesis WP4211', 'Translation Factors WP107', 'Triacylglyceride Synthesis WP325', 'Tryptophan catabolism leading to NAD+ production WP4210', 'Tryptophan metabolism WP465', 'Tumor suppressor activity of SMARCB1 WP4204', 'Type 2 papillary renal cell carcinoma WP4241', 'Type III interferon signaling WP2113', 'Type II diabetes mellitus WP1584', 'Type II interferon signaling (IFNG) WP619', 'Urea cycle and metabolism of amino groups WP497', 'VEGFA-VEGFR2 Signaling Pathway WP3888', 'Valproic acid pathway WP3871', 'Viral Acute Myocarditis WP4298', 'Vitamin A and Carotenoid Metabolism WP716', 'Vitamin B6-dependent and responsive disorders WP4228', 'Vitamin B12 Disorders WP4271', 'Vitamin B12 Metabolism WP1533', 'Vitamin D Metabolism WP1531', 'Vitamin D Receptor Pathway WP2877', 'Vitamin D in inflammatory diseases WP4482', 'White fat cell differentiation WP4149', 'Wnt/beta-catenin Signaling Pathway in Leukemia WP3658', 'Wnt Signaling Pathway WP363', 'Wnt Signaling Pathway and Pluripotency WP399', 'Wnt Signaling WP428', 'Wnt Signaling in Kidney Disease WP4150', 'Zinc homeostasis WP3529', 'exRNA mechanism of action and biogenesis WP2805', 'let-7 inhibition of ES cell reprogramming WP3299', 'mRNA Processing WP411', 'miR-509-3p alteration of YAP1/ECM axis WP3967', 'miR-517 relationship with ARCN1 and USP1 WP3596', 'miRNA Biogenesis WP2338', 'miRNA Regulation of DNA Damage Response WP1530', 'miRNA regulation of p53 pathway in prostate cancer WP3982', 'miRNA regulation of prostate cancer signaling pathways WP3981', 'miRNA targets in ECM and membrane receptors WP2911', 'miRNAs involved in DNA damage response WP1545', 'miRNAs involvement in the immune response in sepsis WP4329', 'miRs in Muscle Cell Differentiation WP2012', 'mir-124 predicted interactions with cell cycle and differentiation  WP3595', 'ncRNAs involved in STAT3 signaling in hepatocellular carcinoma WP4337', 'ncRNAs involved in Wnt signaling in hepatocellular carcinoma WP4336', 'p38 MAPK Signaling Pathway WP400', '', '', '', '', '', '', '', '', '', '', '', ''], [0.0, 0.0, 0.0, 0.00676, 0.0, 0.00241, 0.0, 0.0, 0.00594, 0.0, 0.0, 0.00216, 0.0, 0.0, 0.01073, 0.0, 0.00242, 0.00215, 0.00251, 0.00394, 0.0, 0.00234, 0.00215, 0.0, 0.00218, 0.0, 0.00223, 0.0, 0.0, 0.0, 0.00227, 0.00244, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00258, 0.0, 0.00241, 0.0, 0.0, 0.00385, 0.00378, 0.00382, 0.0, 0.00248, 0.0, 0.0, 0.0, 0.00234, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00262, 0.0, 0.0, 0.0, 0.00846, 0.00521, 0.00256, 0.0, 0.0024, 0.00252, 0.00231, 0.0, 0.0, 0.0, 0.0, 0.00214, 0.0, 0.0, 0.00216, 0.0, 0.0, 0.00412, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00262, 0.00258, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00559, 0.0, 0.00257, 0.00611, 0.0, 0.0, 0.00195, 0.0, 0.0, 0.00628, 0.00228, 0.00194, 0.0, 0.00245, 0.00474, 0.00562, 0.00215, 0.0, 0.0, 0.00251, 0.00246, 0.0, 0.0, 0.0, 0.0, 0.00254, 0.0, 0.00246, 0.0, 0.0, 0.00244, 0.0, 0.0, 0.00241, 0.0, 0.0, 0.00737, 0.00524, 0.0026, 0.00227, 0.00498, 0.0, 0.00242, 0.00246, 0.0, 0.0, 0.0, 0.00215, 0.0, 0.00214, 0.0, 0.00258, 0.00248, 0.0, 0.0, 0.0, 0.00418, 0.0, 0.00685, 0.0, 0.0, 0.00261, 0.00249, 0.0, 0.00253, 0.00482, 0.00238, 0.0, 0.0, 0.0, 0.00216, 0.0, 0.00256, 0.00477, 0.00233, 0.0, 0.0026, 0.0, 0.0, 0.0, 0.00606, 0.0, 0.00226, 0.0, 0.00424, 0.00227, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0024, 0.00236, 0.00233, 0.00241, 0.00239, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00748, 0.00261, 0.00261, 0.0019, 0.0, 0.0021, 0.0, 0.00466, 0.0, 0.00239, 0.00498, 0.0, 0.00258, 0.00258, 0.0023, 0.0, 0.0026, 0.00222, 0.00248, 0.0, 0.0, 0.00426, 0.0, 0.0, 0.00644, 0.0, 0.0, 0.0, 0.0, 0.0025, 0.0, 0.0, 0.0, 0.00246, 0.0, 0.0, 0.0, 0.0, 0.00383, 0.0, 0.0024, 0.0, 0.00717, 0.0, 0.00252, 0.00777, 0.00521, 0.0, 0.0, 0.0, 0.0, 0.00761, 0.0051, 0.00696, 0.00262, 0.0, 0.0, 0.00377, 0.0, 0.0, 0.00522, 0.00259, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00248, 0.0, 0.0, 0.00253, 0.0, 0.00253, 0.0, 0.0, 0.0, 0.0, 0.00377, 0.0, 0.0, 0.00576, 0.00242, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00227, 0.00494, 0.00468, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00245, 0.0046, 0.0, 0.0, 0.0, 0.0, 0.00699, 0.0, 0.00499, 0.00452, 0.0, 0.00216, 0.00225, 0.00242, 0.0, 0.00227, 0.00634, 0.00652, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00249, 0.0025, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00447, 0.0, 0.00222, 0.0, 0.00458, 0.0, 0.00217, 0.0, 0.0, 0.0, 0.0023, 0.0, 0.00537, 0.0019, 0.0, 0.00238, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00262, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00238, 0.00651, 0.0, 0.0, 0.00508, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00249, 0.0, 0.00719, 0.0, 0.0026, 0.0045, 0.0, 0.00236, 0.00563, 0.0, 0.0, 0.0, 0.0, 0.00235, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00512, 0.0, 0.0, 0.0, 0.00197, 0.0, 0.00214, 0.0, 0.0, 0.0, 0.0, 0.00243, 0.0, 0.00259, 0.0, 0.00251, 0.00621, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00246, 0.0026, 0.0, 0.00252, 0.0, 0.00234, 0.0, 0.00512, 0.00475, 0.00246, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00655, 0.0, 0.00218, 0.00239, 0.0, 0.00515, 0.00471, 0.0, 0.0, 0.00504, 0.00246, 0.0, 0.0, 0.0021, 0.00408, 0.00243, 0.00243, 0.0, 0.0, 0.00399, 0.0, 0.0, 0.0, 0.0, 0.00251, 0.0, 0.0, 0.0, 0.0, 0.00248, 0.0, 0.0, 0.00434, 0.00244, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])\n})\n</script>\n"
     },
     "metadata": {}
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "The scaling factor for this canvas is: 0.01073\n"
    }
   ],
   "source": [
    "anneal_list, x_dimension, y_dimension, num_hex = library_processing(0)\n",
    "color_list, scaling_factor = get_color(anneal_list)\n",
    "unzipped_anneal_list = unzip_list(anneal_list)\n",
    "Canvas(x_dimension, y_dimension, color_list, list(unzipped_anneal_list[0]), list(unzipped_anneal_list[2]))\n",
    "print(\"The scaling factor for this canvas is: \" + str(scaling_factor))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/plain": "<IPython.core.display.HTML object>",
      "text/html": "<script src=\"/static/components/requirejs/require.js\"></script>"
     },
     "metadata": {}
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": "<IPython.core.display.HTML object>",
      "text/html": "\n<script>\nrequire.config({\n  paths: {\n    'd3': 'https://cdnjs.cloudflare.com/ajax/libs/d3/5.16.0/d3.min',\n    'd3-hexbin': 'https://d3js.org/d3-hexbin.v0.2.min',\n  },\n  shim: {\n    'd3-hexbin': ['d3']\n  }\n})\n\n// If we configure mychart via url, we can eliminate this define here\ndefine('mychart-87266990-0d00-4490-a2e4-21643c65ea38', ['d3', 'd3-hexbin'], function(d3, d3_hexbin) {\n  return function (figure_id, numA, numB, colorList, libraryList, indices) {\n    var margin = {top: 50, right: 20, bottom: 20, left: 50},\n      width = 850 - margin.left - margin.right,\n      height = 350 - margin.top - margin.bottom;\n\n    // append the svg object to the body of the page\n    var svG = d3.select('#' + figure_id)\n                .attr(\"width\", width + margin.left + margin.right)\n                .attr(\"height\", height + margin.top + margin.bottom)\n                .append(\"g\")\n                .attr(\"transform\", \"translate(\" + margin.left + \",\" + margin.top + \")\");\n\n    //The number of columns and rows of the heatmap\n    var MapColumns = numA,\n        MapRows = numB;\n\n    //The maximum radius the hexagons can have to still fit the screen\n    var hexRadius = d3.min([width/((MapColumns + 0.5) * Math.sqrt(3)), height/((MapRows + 1/3) * 1.5)]);\n\n    //Calculate the center position of each hexagon\n    var points = [];\n    for (var i = 0; i < MapRows; i++) {\n        for (var j = 0; j < MapColumns; j++) {\n            var x = hexRadius * j * Math.sqrt(3)\n            //Offset each uneven row by half of a \"hex-width\" to the right\n            if(i%2 === 1) x += (hexRadius * Math.sqrt(3))/2\n            var y = hexRadius * i * 1.5\n            points.push([x,y])\n        }\n    }\n\n    //Set the hexagon radius\n    var hexbin = d3_hexbin.hexbin().radius(hexRadius);\n\n    svG.append(\"g\")\n      .selectAll(\".hexagon\")\n      .data(hexbin(points))\n      .enter().append(\"path\")\n      .attr(\"class\", \"hexagon\")\n      .attr(\"d\", function (d) {\n          return \"M\" + d.x + \",\" + d.y + hexbin.hexagon();\n      })\n      .attr(\"stroke\", \"white\")\n      .attr(\"stroke-width\", \"1px\")\n      .style(\"fill\", function (d,i) { return colorList[i]; })\n      .on(\"mouseover\", mover)\n      .on(\"mouseout\", mout)\n      .append(\"svg:title\")\n      .text(function(d,i) { return libraryList[i].concat(\" \").concat(indices[i]); });\n\n    // Mouseover function\n    function mover(d) {\n    d3.select(this)\n      .transition().duration(10)  \n      .style(\"fill-opacity\", 0.3)\n    };\n\n    // Mouseout function\n    function mout(d) { \n    d3.select(this)\n      .transition().duration(10)\n      .style(\"fill-opacity\", 1)\n    };\n\n}\n\n})\n</script>\n"
     },
     "metadata": {}
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": "<IPython.core.display.HTML object>",
      "text/html": "\n<svg id='fig-c1e60864-b85f-4feb-8919-4d380ab7c4b0'></svg>\n<script>\nrequire(['mychart-87266990-0d00-4490-a2e4-21643c65ea38'], function(mychart) {\n  mychart('fig-c1e60864-b85f-4feb-8919-4d380ab7c4b0', 22, 22, ['#000000', '#000000', '#3e003e', '#000000', '#000000', '#000000', '#3a003a', '#000000', '#3a003a', '#000000', '#370037', '#000000', '#3a003a', '#000000', '#000000', '#000000', '#6f006f', '#000000', '#000000', '#000000', '#360036', '#350035', '#000000', '#9c009c', '#670067', '#000000', '#000000', '#000000', '#000000', '#000000', '#3c003c', '#000000', '#000000', '#330033', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#970097', '#9b009b', '#390039', '#000000', '#3c003c', '#3e003e', '#000000', '#000000', '#000000', '#910091', '#000000', '#000000', '#ab00ab', '#000000', '#000000', '#3c003c', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#700070', '#000000', '#7c007c', '#860086', '#000000', '#000000', '#000000', '#3a003a', '#000000', '#390039', '#000000', '#3c003c', '#000000', '#000000', '#000000', '#000000', '#370037', '#7c007c', '#000000', '#000000', '#2f002f', '#2d002d', '#7c007c', '#3b003b', '#000000', '#000000', '#000000', '#8d008d', '#760076', '#000000', '#790079', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#990099', '#000000', '#000000', '#000000', '#340034', '#000000', '#3a003a', '#000000', '#000000', '#000000', '#390039', '#380038', '#3a003a', '#000000', '#000000', '#3e003e', '#380038', '#000000', '#360036', '#390039', '#000000', '#000000', '#000000', '#000000', '#3d003d', '#000000', '#000000', '#000000', '#000000', '#5b005b', '#000000', '#330033', '#000000', '#000000', '#000000', '#3a003a', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#380038', '#360036', '#330033', '#000000', '#aa00aa', '#9b009b', '#000000', '#000000', '#000000', '#330033', '#2d002d', '#350035', '#000000', '#390039', '#000000', '#000000', '#3e003e', '#000000', '#000000', '#000000', '#950095', '#3e003e', '#390039', '#000000', '#000000', '#000000', '#370037', '#000000', '#000000', '#000000', '#3c003c', '#000000', '#3e003e', '#000000', '#000000', '#710071', '#3d003d', '#390039', '#000000', '#000000', '#000000', '#000000', '#000000', '#a300a3', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#6d006d', '#750075', '#000000', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#5b005b', '#000000', '#390039', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3d003d', '#000000', '#900090', '#000000', '#3a003a', '#000000', '#390039', '#000000', '#000000', '#000000', '#000000', '#000000', '#620062', '#000000', '#000000', '#000000', '#5a005a', '#330033', '#000000', '#000000', '#000000', '#770077', '#000000', '#000000', '#000000', '#000000', '#380038', '#000000', '#000000', '#2e002e', '#3a003a', '#000000', '#000000', '#000000', '#7a007a', '#000000', '#000000', '#000000', '#000000', '#3c003c', '#3b003b', '#c900c9', '#380038', '#000000', '#000000', '#000000', '#000000', '#000000', '#3a003a', '#000000', '#330033', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#000000', '#b500b5', '#370037', '#000000', '#000000', '#730073', '#3a003a', '#3a003a', '#7a007a', '#000000', '#000000', '#5a005a', '#000000', '#3c003c', '#000000', '#3e003e', '#000000', '#000000', '#000000', '#5f005f', '#3b003b', '#000000', '#3e003e', '#000000', '#af00af', '#360036', '#390039', '#000000', '#790079', '#000000', '#000000', '#630063', '#7a007a', '#000000', '#000000', '#000000', '#000000', '#000000', '#320032', '#000000', '#000000', '#350035', '#000000', '#3b003b', '#000000', '#000000', '#6a006a', '#3e003e', '#000000', '#000000', '#3c003c', '#000000', '#940094', '#000000', '#000000', '#3c003c', '#000000', '#000000', '#330033', '#000000', '#000000', '#000000', '#000000', '#390039', '#000000', '#000000', '#000000', '#000000', '#3a003a', '#3d003d', '#a500a5', '#000000', '#000000', '#710071', '#000000', '#3e003e', '#000000', '#6b006b', '#000000', '#3a003a', '#330033', '#000000', '#000000', '#000000', '#2e002e', '#000000', '#320032', '#000000', '#b900b9', '#860086', '#890089', '#760076', '#3e003e', '#000000', '#000000', '#5e005e', '#7d007d', '#a100a1', '#000000', '#000000', '#390039', '#3a003a', '#000000', '#780078', '#370037', '#ff00ff', '#000000', '#000000', '#000000', '#800080', '#3c003c', '#000000', '#330033', '#3d003d', '#000000', '#5b005b', '#000000', '#000000', '#330033', '#000000', '#6d006d', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3d003d', '#000000', '#360036', '#3c003c', '#000000', '#3a003a', '#000000', '#000000', '#850085', '#360036', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#3e003e', '#000000', '#6b006b', '#3d003d', '#610061', '#650065', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#3b003b', '#000000', '#340034', '#360036', '#000000', '#000000', '#000000', '#650065', '#390039', '#380038', '#000000', '#000000', '#3b003b', '#000000', '#000000', '#000000', '#000000', '#a600a6', '#000000', '#000000', '#000000', '#000000', '#5a005a', '#000000', '#3e003e', '#340034', '#330033', '#000000', '#000000', '#3a003a', '#3e003e', '#3d003d', '#710071', '#000000', '#000000', '#000000', '#000000', '#000000', '#b200b2', '#000000', '#000000', '#000000', '#6f006f', '#350035', '#000000', '#3a003a', '#3a003a', '#000000', '#000000', '#000000', '#000000'], ['Gastric Cancer Network 2 WP2363', 'RAC1/PAK1/p38/MMP2 Pathway WP3303', 'The alternative pathway of fetal androgen synthesis WP4524', 'Signaling of Hepatocyte Growth Factor Receptor WP313', 'Glial Cell Differentiation WP2276', 'ACE Inhibitor Pathway WP554', 'Wnt Signaling in Kidney Disease WP4150', 'Triacylglyceride Synthesis WP325', 'G13 Signaling Pathway WP524', 'Chemokine signaling pathway WP3929', 'IL-4 Signaling Pathway WP395', 'Fluoropyrimidine Activity WP1601', 'Oxidative Stress WP408', 'TGF-B Signaling in Thyroid Cells for Epithelial-Mesenchymal Transition WP3859', 'TFs Regulate miRNAs related to cardiac hypertrophy WP1559', 'Role Altered Glycolysation of MUC1 in Tumour Microenvironment WP4480', 'One carbon metabolism and related pathways WP3940', 'Cysteine and methionine catabolism WP4504', 'exRNA mechanism of action and biogenesis WP2805', 'T-Cell Receptor and Co-stimulatory Signaling WP2583', 'Oncostatin M Signaling Pathway WP2374', 'Arrhythmogenic Right Ventricular Cardiomyopathy WP2118', 'AGE/RAGE pathway WP2324', 'VEGFA-VEGFR2 Signaling Pathway WP3888', 'ncRNAs involved in Wnt signaling in hepatocellular carcinoma WP4336', 'MECP2 and Associated Rett Syndrome WP3584', '', 'GPCRs, Class A Rhodopsin-like WP455', 'Fatty Acid Biosynthesis WP357', 'Small Ligand GPCRs WP247', 'Angiogenesis WP1539', 'EDA Signalling in Hair Follicle Development WP3930', 'Hypothesized Pathways in Pathogenesis of Cardiovascular Disease WP3668', 'Hair Follicle Development: Cytodifferentiation (Part 3 of 3) WP2840', 'Statin Pathway WP430', 'Wnt Signaling Pathway WP363', 'Photodynamic therapy-induced NF-kB survival signaling WP3617', 'Rett syndrome causing genes WP4312', 'MTHFR deficiency WP4288', 'Monoamine GPCRs WP58', 'Pathways Regulating Hippo Signaling WP4540', 'Pathways in clear cell renal cell carcinoma WP4018', 'IL-6 signaling pathway WP364', 'Nucleotide-binding Oligomerization Domain (NOD) pathway WP1433', 'Glycerophospholipid Biosynthetic Pathway WP2533', 'Steroid Biosynthesis WP496', 'Nucleotide GPCRs WP80', 'Spinal Cord Injury WP2431', 'Arachidonate Epoxygenase / Epoxide Hydrolase WP678', 'ESC Pluripotency Pathways WP3931', 'TGF-beta Receptor Signaling WP560', 'ncRNAs involved in STAT3 signaling in hepatocellular carcinoma WP4337', 'Splicing factor NOVA regulated synaptic proteins WP4148', 'Gene regulatory network modelling somitogenesis  WP2854', 'Secretion of Hydrochloric Acid in Parietal Cells WP2597', 'Extracellular vesicles in the crosstalk of cardiac cells WP4300', 'SCFA and skeletal muscle substrate metabolism WP4030', 'Tgif disruption of Shh signaling WP3674', '', 'Aflatoxin B1 metabolism WP699', 'Liver X Receptor Pathway WP2874', 'PI3K/AKT/mTOR - VitD3 Signalling WP4141', 'Serotonin and anxiety-related events WP3944', 'IL-1 signaling pathway WP195', 'Vitamin B12 Metabolism WP1533', '', 'Methylation Pathways WP704', 'Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways WP706', 'Mevalonate pathway WP3963', 'DDX1 as a regulatory component of the Drosha microprocessor WP2942', 'Prostaglandin Synthesis and Regulation WP98', 'Tumor suppressor activity of SMARCB1 WP4204', 'Hepatitis C and Hepatocellular Carcinoma WP3646', 'IL-5 Signaling Pathway WP127', 'miR-509-3p alteration of YAP1/ECM axis WP3967', 'The human immune response to tuberculosis WP4197', 'Blood Clotting Cascade WP272', 'DNA IR-Double Strand Breaks (DSBs) and cellular response via ATM WP3959', 'Homologous recombination WP186', 'RalA downstream regulated genes WP2290', 'Hematopoietic Stem Cell Differentiation WP2849', 'NAD Biosynthesis II (from tryptophan) WP2485', 'Nanoparticle triggered autophagic cell death WP2509', 'ID signaling pathway WP53', 'TGF-beta Signaling Pathway WP366', 'Integrated Breast Cancer Pathway WP1984', 'Circadian rhythm related genes WP3594', 'Physiological and Pathological Hypertrophy  of the Heart WP1528', 'Deregulation of Rab and Rab Effector Genes in Bladder Cancer WP2291', 'Phase I biotransformations, non P450 WP136', 'Pathogenic Escherichia coli infection WP2272', 'Adipogenesis WP236', 'Follicle Stimulating Hormone (FSH) signaling pathway WP2035', 'Serotonin and anxiety WP3947', 'Serotonin Receptor 4/6/7 and NR3C Signaling WP734', 'Preimplantation Embryo WP3527', 'IL-7 Signaling Pathway WP205', 'Pregnane X Receptor pathway WP2876', 'Degradation pathway of sphingolipids, including diseases WP4153', 'Prader-Willi and Angelman Syndrome WP3998', 'Oxidation by Cytochrome P450 WP43', 'Ovarian Infertility Genes WP34', 'MAPK Signaling Pathway WP382', 'TLR4 Signaling and Tolerance WP3851', 'Vitamin B6-dependent and responsive disorders WP4228', 'Melatonin metabolism and effects WP3298', 'Pyrimidine metabolism WP4022', 'Neural Crest Differentiation WP2064', 'Target Of Rapamycin (TOR) Signaling WP1471', 'FTO Obesity Variant Mechanism WP3407', 'LncRNA-mediated mechanisms of therapeutic resistance WP3672', 'DNA Replication WP466', 'Bladder Cancer WP2828', 'Structural Pathway of Interleukin 1 (IL-1) WP2637', 'p38 MAPK Signaling Pathway WP400', 'BMP Signaling Pathway in Eyelid Development WP3927', '', 'Insulin signalling in human adipocytes (diabetic condition) WP3635', 'Synaptic signaling pathways associated with autism spectrum disorder WP4539', 'Non-genomic actions of 1,25 dihydroxyvitamin D3 WP4341', 'Endochondral Ossification WP474', 'ATM Signaling Pathway WP2516', 'Farnesoid X Receptor  Pathway WP2879', 'Transcriptional cascade regulating adipogenesis WP4211', 'Toll-like Receptor Signaling Pathway WP75', 'Toll-like Receptor Signaling WP3858', \"ERK Pathway in Huntington's Disease WP3853\", 'Type 2 papillary renal cell carcinoma WP4241', 'SREBF and miR33 in cholesterol and lipid homeostasis WP2011', 'Phosphodiesterases in neuronal function WP4222', 'H19 action Rb-E2F1 signaling and CDK-Beta-catenin activity WP3969', 'Mesodermal Commitment Pathway WP2857', '', 'Pancreatic adenocarcinoma pathway WP4263', 'Simplified Interaction Map Between LOXL4 and Oxidative Stress Pathway WP3670', 'Heart Development WP1591', 'Differentiation of white and brown adipocyte    WP2895', 'GABA receptor Signaling WP4159', 'Photodynamic therapy-induced HIF-1 survival signaling WP3614', 'Urea cycle and metabolism of amino groups WP497', 'Leptin Insulin Overlap WP3935', 'Osteoblast Signaling WP322', 'Acetylcholine Synthesis WP528', 'Mammary gland development pathway - Puberty (Stage 2 of 4) WP2814', 'Thiamine metabolic pathways WP4297', 'IL-3 Signaling Pathway WP286', 'Folate Metabolism WP176', 'Cytoplasmic Ribosomal Proteins WP477', 'Inflammatory Response Pathway WP453', 'Metapathway biotransformation Phase I and II WP702', 'Selenium Micronutrient Network WP15', 'Non-homologous end joining WP438', 'Sulindac Metabolic Pathway WP2542', 'Macrophage markers WP4146', 'ErbB Signaling Pathway WP673', 'Regulation of Actin Cytoskeleton WP51', 'Parkin-Ubiquitin Proteasomal System pathway WP2359', 'Regulation of Apoptosis by Parathyroid Hormone-related Protein WP3872', 'Metabolic reprogramming in colon cancer WP4290', 'NAD+ metabolism WP3644', 'Development of pulmonary dendritic cells and macrophage subsets WP3892', 'Folate-Alcohol and Cancer Pathway Hypotheses WP1589', 'Drug Induction of Bile Acid Pathway WP2289', 'PDGF Pathway WP2526', 'Initiation of transcription and translation elongation at the HIV-1 LTR WP3414', 'Electron Transport Chain (OXPHOS system in mitochondria) WP111', 'Leptin and adiponectin WP3934', 'Regulation of Microtubule Cytoskeleton WP2038', 'Osteoclast Signaling WP12', 'Eicosanoid Synthesis WP167', 'Nicotine Metabolism WP1600', 'Complement and Coagulation Cascades WP558', 'Biogenic Amine Synthesis WP550', 'Dual hijack model of Vif in HIV infection WP3300', 'Serotonin Transporter Activity WP1455', 'Methionine De Novo and Salvage Pathway WP3580', 'let-7 inhibition of ES cell reprogramming WP3299', 'NAD metabolism, sirtuins and aging WP3630', 'Type II interferon signaling (IFNG) WP619', 'DNA Mismatch Repair WP531', 'Tryptophan metabolism WP465', 'Dopamine metabolism WP2436', 'Common Pathways Underlying Drug Addiction WP2636', 'EPO Receptor Signaling WP581', 'IL-10 Anti-inflammatory Signaling Pathway  WP4495', 'Metabolic pathway of LDL, HDL and TG, including diseases WP4522', 'Globo Sphingolipid Metabolism WP1424', 'Imatinib and Chronic Myeloid Leukemia WP3640', 'Genotoxicity pathway WP4286', 'NRF2-ARE regulation WP4357', 'GPR40 Pathway WP3958', 'Role of Osx and miRNAs in tooth development WP3971', 'MET in type 1 papillary renal cell carcinoma WP4205', 'MicroRNA for Targeting Cancer Growth and Vascularization in Glioblastoma WP3593', 'BMP2-WNT4-FOXO1 Pathway in Human Primary Endometrial Stromal Cell Differentiation WP3876', 'Proteasome Degradation WP183', 'One Carbon Metabolism WP241', 'Cell Differentiation - Index expanded WP2023', 'MicroRNAs in cardiomyocyte hypertrophy WP1544', 'Wnt/beta-catenin Signaling Pathway in Leukemia WP3658', '', 'Alpha 6 Beta 4 signaling pathway WP244', 'Gastric Cancer Network 1 WP2361', 'Calcium Regulation in the Cardiac Cell WP536', 'Genes targeted by miRNAs in adipocytes WP1992', 'IL-2 Signaling Pathway WP49', 'Cytokines and Inflammatory Response WP530', 'Ethanol metabolism resulting in production of ROS by CYP2E1 WP4269', 'Regulation of sister chromatid separation at the metaphase-anaphase transition WP4240', 'Retinoblastoma Gene in Cancer WP2446', 'Metabolism of Spingolipids in ER and Golgi apparatus WP4142', '', 'Bile Acids synthesis and enterohepatic circulation  WP4389', 'Nanomaterial induced apoptosis WP2507', 'Hippo-Merlin Signaling Dysregulation WP4541', 'miRNA regulation of prostate cancer signaling pathways WP3981', 'White fat cell differentiation WP4149', 'Fibrin Complement Receptor 3 Signaling Pathway WP4136', 'Ferroptosis WP4313', 'Nucleotide Metabolism WP404', 'miRNA Regulation of DNA Damage Response WP1530', 'Effects of Nitric Oxide WP1995', 'ATR Signaling WP3875', '', 'DNA Damage Response (only ATM dependent) WP710', 'Mechanoregulation and pathology of YAP/TAZ via Hippo and non-Hippo mechanisms WP4534', 'Hippo-Yap signaling pathway WP4537', 'Non-small cell lung cancer WP4255', 'Nonalcoholic fatty liver disease WP4396', 'G Protein Signaling Pathways WP35', 'Cholesterol Biosynthesis Pathway WP197', 'IL1 and megakaryocytes in obesity WP2865', 'Gastric acid production WP2596', 'PPAR Alpha Pathway WP2878', 'Cells and Molecules involved in local acute inflammatory response  WP4493', 'Sleep regulation WP3591', 'Exercise-induced Circadian Regulation WP410', 'DNA Damage Response WP707', 'Cardiac Progenitor Differentiation WP2406', 'Estrogen Receptor Pathway WP2881', 'DNA IR-damage and cellular response via ATR WP4016', 'Ectoderm Differentiation WP2858', 'BDNF-TrkB Signaling WP3676', 'Arylamine metabolism WP694', 'GPCRs, Class C Metabotropic glutamate, pheromone WP501', 'Metastatic brain tumor WP2249', 'Vitamin B12 Disorders WP4271', 'Striated Muscle Contraction Pathway WP383', 'Cori Cycle WP1946', 'GPCRs, Class B Secretin-like WP334', '4-hydroxytamoxifen, Dexamethasone, and Retinoic Acids Regulation of p27 Expression WP3879', 'Neovascularisation processes WP4331', 'Cannabinoid receptor signaling WP3869', 'Ciliary landscape WP4352', 'Translation Factors WP107', 'Mammary gland development pathway - Embryonic development (Stage 1 of 4) WP2813', 'Lung fibrosis WP3624', 'EV release from cardiac cells and their functional effects WP3297', 'NRF2 pathway WP2884', 'Composition of Lipid Particles WP3601', 'Factors and pathways affecting insulin-like growth factor (IGF1)-Akt signaling WP3850', 'Mammary gland development pathway - Involution (Stage 4 of 4) WP2815', 'GPCRs, Other WP117', 'G1 to S cell cycle control WP45', 'Serotonin Receptor 2 and ELK-SRF/GATA4 signaling WP732', 'PI3K-AKT-mTOR signaling pathway and therapeutic opportunities WP3844', 'Sulfation Biotransformation Reaction WP692', 'miRs in Muscle Cell Differentiation WP2012', 'SRF and miRs in Smooth Muscle Differentiation and Proliferation WP1991', 'TCA Cycle (aka Krebs or citric acid cycle) WP78', 'Mitochondrial Gene Expression WP391', 'Kit receptor signaling pathway WP304', 'Thymic Stromal LymphoPoietin (TSLP) Signaling Pathway WP2203', 'Purine metabolism WP4224', 'Glycogen Synthesis and Degradation WP500', 'Fatty Acid Beta Oxidation WP143', 'Ethanol effects on histone modifications WP3996', 'TCA Cycle and Deficiency of Pyruvate Dehydrogenase complex (PDHc) WP2453', 'Canonical  and Non-canonical Notch signaling WP3845', 'Monoamine Transport WP727', 'Myometrial Relaxation and Contraction Pathways WP289', '', 'miRNA regulation of p53 pathway in prostate cancer WP3982', 'Peptide GPCRs WP24', 'Molybdenum cofactor (Moco) biosynthesis WP4507', 'Vitamin D Receptor Pathway WP2877', 'Signaling Pathways in Glioblastoma WP2261', 'Platelet-mediated interactions with vascular and circulating cells WP4462', 'mRNA Processing WP411', 'Lipid Metabolism Pathway WP3965', 'Somatroph axis (GH) and its relationship to dietary restriction and aging WP4186', 'Disorders of the Krebs cycle WP4236', 'ATM Signaling Network in Development and Disease  WP3878', 'Focal Adhesion-PI3K-Akt-mTOR-signaling pathway WP3932', 'Pathways Affected in Adenoid Cystic Carcinoma WP3651', 'Vitamin A and Carotenoid Metabolism WP716', 'Codeine and Morphine Metabolism WP1604', 'Mitochondrial LC-Fatty Acid Beta-Oxidation WP368', 'PTF1A related regulatory pathway WP4147', 'Vitamin D Metabolism WP1531', 'Genes related to primary cilium development (based on CRISPR) WP4536', 'Tryptophan catabolism leading to NAD+ production WP4210', '', 'Synthesis and Degradation of Ketone Bodies WP311', 'Overview of nanoparticle effects WP3287', 'Matrix Metalloproteinases WP129', 'Glutathione metabolism WP100', 'Integrin-mediated Cell Adhesion WP185', 'TP53 Network WP1742', 'Interactome of polycomb repressive complex 2 (PRC2)  WP2916', 'Leptin signaling pathway WP2034', 'Benzene metabolism WP3891', 'Photodynamic therapy-induced unfolded protein response WP3613', 'Interleukin-1 Induced Activation of NF-kappa-B WP3656', 'miR-517 relationship with ARCN1 and USP1 WP3596', 'Primary Focal Segmental Glomerulosclerosis FSGS WP2572', 'Trans-sulfuration pathway WP2333', 'Benzo(a)pyrene metabolism WP696', 'Photodynamic therapy-induced NFE2L2 (NRF2) survival signaling WP3612', 'Estrogen signaling pathway WP712', 'Microglia Pathogen Phagocytosis Pathway WP3937', 'Thermogenesis WP4321', 'ApoE and miR-146 in inflammation and atherosclerosis WP3926', 'EBV LMP1 signaling WP262', 'Transcription factor regulation in adipogenesis WP3599', 'Senescence and Autophagy in Cancer WP615', 'Regulation of toll-like receptor signaling pathway WP1449', 'Corticotropin-releasing hormone signaling pathway WP2355', 'Gamma-Glutamyl Cycle for the biosynthesis and degradation of glutathione, including diseases WP4518', 'Tamoxifen metabolism WP691', 'Oligodendrocyte Specification and differentiation(including remyelination), leading to Myelin Components for CNS WP4304', 'IL17 signaling pathway WP2112', 'Interleukin-11 Signaling Pathway WP2332', 'Development and heterogeneity of the ILC family WP3893', 'PDGFR-beta pathway WP3972', 'Simplified Depiction of MYD88 Distinct Input-Output Pathway WP3877', 'Hereditary leiomyomatosis and renal cell carcinoma pathway WP4206', 'Mammary gland development pathway - Pregnancy and lactation (Stage 3 of 4) WP2817', 'Classical pathway of steroidogenesis, including diseases WP4523', 'Mitochondrial complex I assembly model OXPHOS system WP4324', 'Disorders of Folate Metabolism and Transport WP4259', 'Cell Differentiation - Index WP2029', 'Energy Metabolism WP1541', 'NO/cGMP/PKG mediated Neuroprotection WP4008', 'Cerebral Organic Acidurias, including diseases WP4519', 'Pentose Phosphate Metabolism WP134', 'PPAR signaling pathway WP3942', 'Fas Ligand (FasL) pathway and Stress induction of Heat Shock Proteins (HSP) regulation WP314', 'Amyotrophic lateral sclerosis (ALS) WP2447', 'TNF alpha Signaling Pathway WP231', 'miRNAs involvement in the immune response in sepsis WP4329', 'Bone Morphogenic Protein (BMP) Signalling and Regulation WP1425', 'Canonical and Non-Canonical TGF-B signaling WP3874', 'Endoderm Differentiation WP2853', 'Osteopontin Signaling WP1434', 'Wnt Signaling Pathway and Pluripotency WP399', 'IL-9 Signaling Pathway WP22', 'Methionine metabolism leading to Sulphur Amino Acids and related disorders WP4292', 'Epithelial to mesenchymal transition in colorectal cancer WP4239', 'Nuclear Receptors Meta-Pathway WP2882', 'Intraflagellar transport proteins binding to dynein WP4532', 'Robo4 and VEGF Signaling Pathways Crosstalk WP3943', 'TNF related weak inducer of apoptosis (TWEAK) Signaling Pathway WP2036', 'Hypertrophy Model WP516', 'Angiopoietin Like Protein 8 Regulatory Pathway WP3915', 'Focal Adhesion WP306', 'AMP-activated Protein Kinase (AMPK) Signaling WP1403', 'Thyroxine (Thyroid Hormone) Production WP1981', 'Transcription co-factors SKI and SKIL protein partners WP4533', 'Selenium Metabolism and Selenoproteins WP28', 'Parkinsons Disease Pathway WP2371', 'Synaptic Vesicle Pathway WP2267', 'Vitamin D in inflammatory diseases WP4482', 'RIG-I-like Receptor Signaling WP3865', 'Amino Acid metabolism WP3925', 'Oxysterols derived from cholesterol WP4545', 'T-Cell antigen Receptor (TCR)  Signaling Pathway WP69', 'Signal transduction through IL1R WP4496', 'Ras Signaling WP4223', 'Complement Activation WP545', 'Integrated Cancer Pathway WP1971', 'Androgen receptor signaling pathway WP138', 'Ganglio Sphingolipid Metabolism WP1423', 'Hypothetical Craniofacial Development Pathway WP3655', 'Brain-Derived Neurotrophic Factor (BDNF) signaling pathway WP2380', 'Cardiac Hypertrophic Response WP2795', 'B Cell Receptor Signaling Pathway WP23', 'Allograft Rejection WP2328', 'RANKL/RANK (Receptor activator of NFKB (ligand)) Signaling Pathway WP2018', 'Oxidative phosphorylation WP623', 'Chromosomal and microsatellite instability in colorectal cancer  WP4216', 'Endometrial cancer WP4155', 'Alzheimers Disease WP2059', 'HIF1A and PPARG regulation of glycolysis WP2456', 'Supression of HMGB1 mediated inflammation by THBD WP4479', 'Prion disease pathway WP3995', 'Caloric restriction and aging WP4191', 'NAD+ biosynthetic pathways WP3645', 'Kennedy pathway from Sphingolipids WP3933', 'Extracellular vesicle-mediated signaling in recipient cells WP2870', 'Association Between Physico-Chemical Features and Toxicity Associated Pathways WP3680', 'Nicotine Activity on Dopaminergic Neurons WP1602', 'TYROBP Causal Network WP3945', 'Zinc homeostasis WP3529', 'Alanine and aspartate metabolism WP106', 'miRNAs involved in DNA damage response WP1545', 'EGF/EGFR Signaling Pathway WP437', 'Human Thyroid Stimulating Hormone (TSH) signaling pathway WP2032', 'Aryl Hydrocarbon Receptor Pathway WP2873', 'Aryl Hydrocarbon Receptor WP2586', 'MAPK Cascade WP422', 'Eukaryotic Transcription Initiation WP405', 'Photodynamic therapy-induced AP-1 survival signaling. WP3611', 'mir-124 predicted interactions with cell cycle and differentiation  WP3595', 'Cell-type Dependent Selectivity of CCK2R Signaling WP3679', 'The effect of progerin on the involved genes in Hutchinson-Gilford Progeria Syndrome WP4320', 'Apoptosis Modulation by HSP70 WP384', 'Phytochemical activity on NRF2 transcriptional activation WP3', 'Nanoparticle-mediated activation of receptor signaling WP2643', 'Insulin signalling in human adipocytes (normal condition) WP3634', 'Glycosylation and related congenital defects. WP4521', 'Sterol Regulatory Element-Binding Proteins (SREBP) signalling WP1982', 'Iron metabolism in placenta WP2007', 'Wnt Signaling WP428', 'LncRNA involvement in canonical Wnt signaling and colorectal cancer WP4258', 'miRNA targets in ECM and membrane receptors WP2911', 'Nanoparticle triggered regulated necrosis WP2513', 'Notch Signaling Pathway WP61', 'Type II diabetes mellitus WP1584', 'Constitutive Androstane Receptor Pathway WP2875', 'Hfe effect on hepcidin production WP3924', 'TCA Cycle Nutrient Utilization and Invasiveness of Ovarian Cancer WP2868', 'Dopaminergic Neurogenesis WP2855', 'Sphingolipid pathway WP1422', 'Copper homeostasis WP3286', 'Apoptosis WP254', 'Histone Modifications WP2369', 'Notch Signaling WP268', 'Nuclear Receptors in Lipid Metabolism and Toxicity WP299', 'MFAP5-mediated ovarian cancer cell motility and invasiveness WP3301', 'Human Complement System WP2806', 'Glycolysis and Gluconeogenesis WP534', 'Apoptosis-related network due to altered Notch3 in ovarian cancer WP2864', 'Pyrimidine metabolism and related diseases WP4225', 'Oxidative Damage WP3941', 'Glucuronidation WP698', 'Differentiation Pathway WP2848', 'Type III interferon signaling WP2113', 'NLR Proteins WP288', '', 'PI3K-Akt Signaling Pathway WP4172', 'Valproic acid pathway WP3871', 'NOTCH1 regulation of human endothelial cell calcification WP3413', 'Estrogen metabolism WP697', 'Fatty Acid Omega Oxidation WP206', 'Breast cancer pathway WP4262', 'Regulation of Wnt/B-catenin Signaling by Small Molecule Compounds WP3664', 'Heme Biosynthesis WP561', 'Viral Acute Myocarditis WP4298', 'Apoptosis Modulation and Signaling WP1772', 'Cytosine methylation WP3585', 'MAPK  and NFkB Signalling Pathways Inhibited by Yersinia YopJ WP3849', 'Nuclear Receptors WP170', 'Glucocorticoid and Mineralcorticoid Metabolism WP237', 'Hedgehog Signaling Pathway WP47', 'Hedgehog Signaling Pathway WP4249', '', 'Irinotecan Pathway WP229', 'miRNA Biogenesis WP2338', 'Ebola Virus Pathway on Host WP4217', 'Catalytic cycle of mammalian Flavin-containing MonoOxygenases (FMOs) WP688', 'Insulin Signaling WP481', 'Selective expression of chemokine receptors during T-cell polarization WP4494', 'Signal Transduction of S1P Receptor WP26', 'Hematopoietic Stem Cell Gene Regulation by GABP alpha/beta Complex WP3657', 'Interferon type I signaling pathways WP585', 'Prolactin Signaling Pathway WP2037', 'Amplification and Expansion of Oncogenic Pathways as Metastatic Traits WP3678', 'Endothelin Pathways WP2197', 'Trans-sulfuration and one carbon metabolism WP2525', 'Inhibition of exosome biogenesis and secretion by Manumycin A in CRPC cells WP4301', 'T-Cell antigen Receptor (TCR) pathway during Staphylococcus aureus infection WP3863', 'Cell Cycle WP179', 'Resistin as a regulator of inflammation WP4481'], [0.0, 0.0, 0.00259, 0.0, 0.0, 0.0, 0.00243, 0.0, 0.00242, 0.0, 0.00233, 0.0, 0.00245, 0.0, 0.0, 0.0, 0.00468, 0.0, 0.0, 0.0, 0.00227, 0.00223, 0.0, 0.00655, 0.00434, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00251, 0.0, 0.0, 0.00216, 0.0, 0.0, 0.0, 0.0, 0.0025, 0.0, 0.00634, 0.00652, 0.00239, 0.0, 0.00253, 0.0026, 0.0, 0.0, 0.0, 0.00611, 0.0, 0.0, 0.00719, 0.0, 0.0, 0.00254, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00471, 0.0, 0.00521, 0.00563, 0.0, 0.0, 0.0, 0.00246, 0.0, 0.00241, 0.0, 0.00251, 0.0, 0.0, 0.0, 0.0, 0.00233, 0.00522, 0.0, 0.0, 0.00197, 0.0019, 0.00521, 0.0025, 0.0, 0.0, 0.0, 0.00594, 0.00498, 0.0, 0.00508, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00644, 0.0, 0.0, 0.0, 0.00217, 0.0, 0.00243, 0.0, 0.0, 0.0, 0.00241, 0.00236, 0.00244, 0.0, 0.0, 0.00261, 0.00235, 0.0, 0.00228, 0.00241, 0.0, 0.0, 0.0, 0.0, 0.00257, 0.0, 0.0, 0.0, 0.0, 0.00383, 0.0, 0.00216, 0.0, 0.0, 0.0, 0.00246, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00236, 0.00227, 0.00216, 0.0, 0.00717, 0.00651, 0.0, 0.0, 0.0, 0.00215, 0.0019, 0.00225, 0.0, 0.0024, 0.0, 0.0, 0.0026, 0.0, 0.0, 0.0, 0.00628, 0.0026, 0.00238, 0.0, 0.0, 0.0, 0.00231, 0.0, 0.0, 0.0, 0.00252, 0.0, 0.00259, 0.0, 0.0, 0.00475, 0.00258, 0.0024, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00685, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00458, 0.00494, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00248, 0.00382, 0.0, 0.0024, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00258, 0.0, 0.00606, 0.0, 0.00246, 0.0, 0.00241, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00412, 0.0, 0.0, 0.0, 0.00377, 0.00214, 0.0, 0.0, 0.0, 0.00499, 0.0, 0.0, 0.0, 0.0, 0.00234, 0.0, 0.0, 0.00195, 0.00244, 0.0, 0.0, 0.0, 0.00515, 0.0, 0.0, 0.0, 0.0, 0.00253, 0.00248, 0.00846, 0.00234, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00246, 0.0, 0.00215, 0.0, 0.0, 0.0, 0.0, 0.00248, 0.0, 0.0, 0.00761, 0.0023, 0.0, 0.0, 0.00482, 0.00244, 0.00246, 0.00512, 0.0, 0.0, 0.00377, 0.0, 0.00251, 0.0, 0.00262, 0.0, 0.0, 0.0, 0.00399, 0.00248, 0.0, 0.00262, 0.0, 0.00737, 0.00227, 0.00239, 0.0, 0.0051, 0.0, 0.0, 0.00418, 0.00512, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0021, 0.0, 0.0, 0.00222, 0.0, 0.00249, 0.0, 0.0, 0.00447, 0.0026, 0.0, 0.0, 0.00251, 0.0, 0.00621, 0.0, 0.0, 0.00252, 0.0, 0.0, 0.00214, 0.0, 0.0, 0.0, 0.0, 0.00239, 0.0, 0.0, 0.0, 0.0, 0.00246, 0.00256, 0.00696, 0.0, 0.0, 0.00474, 0.0, 0.00262, 0.0, 0.00452, 0.0, 0.00242, 0.00214, 0.0, 0.0, 0.0, 0.00194, 0.0, 0.0021, 0.0, 0.00777, 0.00562, 0.00576, 0.00498, 0.00262, 0.0, 0.0, 0.00394, 0.00524, 0.00676, 0.0, 0.0, 0.00238, 0.00242, 0.0, 0.00504, 0.0023, 0.01073, 0.0, 0.0, 0.0, 0.00537, 0.00252, 0.0, 0.00215, 0.00258, 0.0, 0.00385, 0.0, 0.0, 0.00216, 0.0, 0.0046, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00258, 0.0, 0.00227, 0.00253, 0.0, 0.00243, 0.0, 0.0, 0.00559, 0.00227, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00248, 0.00261, 0.0, 0.0045, 0.00258, 0.00408, 0.00426, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00249, 0.0, 0.00218, 0.00226, 0.0, 0.0, 0.0, 0.00424, 0.00238, 0.00234, 0.0, 0.0, 0.00249, 0.0, 0.0, 0.0, 0.0, 0.00699, 0.0, 0.0, 0.0, 0.0, 0.00378, 0.0, 0.0026, 0.00218, 0.00215, 0.0, 0.0, 0.00242, 0.00261, 0.00256, 0.00477, 0.0, 0.0, 0.0, 0.0, 0.0, 0.00748, 0.0, 0.0, 0.0, 0.00466, 0.00222, 0.0, 0.00245, 0.00246, 0.0, 0.0, 0.0, 0.0])\n})\n</script>\n"
     },
     "metadata": {}
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "The scaling factor for this canvas is: 0.01073\n"
    }
   ],
   "source": [
    "anneal_list, x_dimension, y_dimension, num_hex = library_processing(0)\n",
    "anneal_list = annealing(anneal_list, 5000, find_fitness(anneal_list)[0])\n",
    "color_list, scaling_factor = get_color(anneal_list)\n",
    "unzipped_anneal_list = unzip_list(anneal_list)\n",
    "Canvas(x_dimension, y_dimension, color_list, list(unzipped_anneal_list[0]), list(unzipped_anneal_list[2]))\n",
    "print(\"The scaling factor for this canvas is: \" + str(scaling_factor))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7.6 64-bit ('base': conda)",
   "language": "python",
   "name": "python37664bitbasecondad81f4d096e8b4b9885e3d702e0bb3be1"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6-final"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}